src/gsd.js

```diff
@@ -100,8 +101,8 @@ function gsd(x, yIn, options) {
     // By the intermediate value theorem We cannot find 2 consecutive maximum or minimum
     for (let i = 1; i < Y.length - 1; ++i) {
         // Minimum in first derivative
-        if ((dY[i] < dY[i - 1]) && (dY[i] <= dY[i + 1]) ||
-            (dY[i] <= dY[i - 1]) && (dY[i] < dY[i + 1])) {
+        if ((dY[i] - dY[i - 1] <  -options.derivativeThreshold) && (dY[i] - dY[i + 1] <= -options.derivativeThreshold) ||
```

Comment:
I don't think this is what I had in mind. Because very high values in the first derivative, but that does not change change in the top, which is usually very flat, will not count as peaks. My idea was to exclude the peaks that are smaller than the derivativeThreshold.
`` if(Math.abs(dY[i])<options.derivativeThreshold)``
The condition you add is more like a threshold in the second derivative, which I spoke about too.